### ISSUE TITLE
Misplaced colon after auto completing a field

### ISSUE THREAD

**Issue by Riateche** [intellij-rust/intellij-rust#5875]
## Environment

* **IntelliJ Rust plugin version:** 0.3.129.3266-202-nightly
* **Rust toolchain version:** stable
* **IDE name and version:** CLion 2020.2
* **Operating system:** Linux

## Problem description

Auto completion inserts a misplaced colon.

## Steps to reproduce
```rust
struct A {
    field1: u32,
    field2: u32,
}

impl A {
    fn f(&self) -> A {
        A { field1: self.fi }
    }
}
```
Place cursor after `self.fi`, press Ctrl+Space (auto complete) and select `field1`. The result is:
```rust
        A { field1: self.field1:  }
```
The expected result is:
```rust
        A { field1: self.field1 }
```

**Comment by Undin** [comment]
I suppose that the bug was introduced in #5659
CC @Kobzol

**Comment by Riateche** [comment]
> adds : after completed struct field literal if no matching (same name and same type) local binding is found.

By the way, that behavior is harmful. It's not immediately obvious if there is a matching variable in scope, especially taking the type in account (which heavily depends on whether the plugin understands types correctly in this context). After completing a suggestion, I want to immediately continue typing, and to do that I need to have a clear idea of what is going to be inserted. So usually if I want to initialize a field, I accept the suggestion and immediately input a colon (or a comma, if I intend to use the shorthand syntax). If the inserted text changes arbitrarily, I'm forced to make a pause after accepting a suggestion, which is a huge waste of time. I think rolling back that change is the only acceptable solution.

**Comment by Kobzol** [comment]
Hmm, I didn't think of that. I agree that maybe it is not ideal that the completion doesn't behave in the same way all the time. And on top of that, now I realize that we can't actually predict whether the user really wanted to use the shorthand syntax (even if there is a matching binding in scope). @Undin should we revert #5659?

**Comment by Undin** [comment]
@Kobzol yep, if we don't know a simple way how to fix the bug (especially, after release branch creation 😄 ), it's better to revert the corresponding changes and try again in separate PR (if possible, of course).

**Comment by Kobzol** [comment]
Well we could fix the bug, but what @Riateche mentioned leads me to believe that the feature itself is maybe not such a good idea, so maybe we shouldn't even include it at all. Should I send the revert PR against `master` or `release-128`?

**Comment by Undin** [comment]
> Should I send the revert PR against master or release-128?

@Kobzol to master to fix the corresponding issue for users with nightly plugin. We will cherry-pick it into `release-128` after merge

**Comment by Kobzol** [comment]
@Undin can we close this issue?

**Comment by Undin** [comment]
Fixed in #5879
